cmd/kube-apiserver/app/aggregator.go

```diff
@@ -211,8 +211,8 @@ func makeAPIServiceAvailableHealthCheck(name string, apiServices []*apiregistrat
 
 	// Watch add/update events for APIServices
 	apiServiceInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { handleAPIServiceChange(obj.(*apiregistration.APIService)) },
-		UpdateFunc: func(old, new interface{}) { handleAPIServiceChange(new.(*apiregistration.APIService)) },
+		AddFunc:    func(obj interface{}, rpId string) { handleAPIServiceChange(obj.(*apiregistration.APIService)) },
```

Comment:
what is the name format of the rpId, they need to be unique so a guid string probably is needed.

Reply:
Possible. Right now, we will use 0,1,2 as the sequence they come from KubeConfigFile arg. They only need to be unique to each other inside a scheduler.